sql/src/main/java/io/crate/expression/symbol/Literal.java

```diff
@@ -158,6 +158,24 @@ public boolean canBeCasted() {
         return true;
     }
 
+    /**
+     * Checks whether this Literal can be converted to the given
+     * DataType without loss of information.
+     * @param dataType The type to check for lossless conversion.
+     * @return True if lossless conversion is possible, false otherwise.
+     */
+    public boolean isLosslesslyConvertableTo(DataType dataType) {
+        if (!type.isConvertableTo(dataType)) {
+            return false;
+        }
+        try {
+            dataType.value(this.value);
+        } catch (IllegalArgumentException | ClassCastException e) {
+            return false;
```

Comment:
Thinking about this some more I think we should do this in a way that avoids Exceptions for control flow handling. generating exceptions is rather expensive.

Reply:
Not sure how expensive it is here because it only affects Literals. We could return `null` if the conversion fails.

Reply:
The problem with null is that we use it currently use it to indicate `NULL` which are two different concepts.